WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -0,0 +1,267 @@
+        formattedPrice(order?.totalTax, currency: order?.currency)
+    }
+
+    // MARK: - View states
+
+    var showRecalculateButton: Bool {
+        !areAmountsFullyCalculated &&
+        isSyncingOrder == false
+    }
+
+    var areAmountsFullyCalculated: Bool {
+        isSyncingOrder == false &&
+        formattedOrderTotalTaxPrice != nil &&
+        formattedOrderTotalPrice != nil
+    }
+
+    var isShimmering: Bool {
+        isSyncingOrder
+    }
+
+    var isPriceFieldRedacted: Bool {
+        formattedOrderTotalTaxPrice == nil || isSyncingOrder
+    }
+
+    var isTotalPriceFieldRedacted: Bool {
+        formattedOrderTotalPrice == nil || isSyncingOrder
+    }
+
+    private let orderService: POSOrderServiceProtocol
+    private let cardPresentPaymentService: CardPresentPaymentFacade
+    private let currencyFormatter: CurrencyFormatter
+
+    init(orderService: POSOrderServiceProtocol,
+         cardPresentPaymentService: CardPresentPaymentFacade,
+         currencyFormatter: CurrencyFormatter) {
+        self.orderService = orderService
+        self.cardPresentPaymentService = cardPresentPaymentService
+        self.currencyFormatter = currencyFormatter
+
+        observeConnectedReaderForStatus()
+        observeCardPresentPaymentEvents()
+    }
+
+    func calculateAmountsTapped(with cartItems: [CartItem], allItems: [POSItem]) {
+        startSyncingOrder(with: cartItems, allItems: allItems)
+    }
+
+    func startSyncingOrder(with cartItems: [CartItem], allItems: [POSItem]) {
+        Task { @MainActor in
+            calculateCartTotal(cartItems: cartItems)
+            await syncOrder(for: cartItems, allItems: allItems)
+        }
+    }
+
+    func cardPaymentTapped() {
+        Task { @MainActor in
+            await collectPayment()
+        }
+    }
+
+    func startNewTransaction() {
+        paymentState = .acceptingCard
+        clearOrder()
+    }
+
+    @MainActor
+    func onTotalsViewDisappearance() {
+        cardPresentPaymentService.cancelPayment()
+    }
+}
+
+// MARK: - Order syncing
+
+private extension TotalsViewModel {
+    @MainActor
+    func syncOrder(for cartProducts: [CartItem], allItems: [POSItem]) async {
+        guard isSyncingOrder == false else {
+            return
+        }
+        isSyncingOrder = true
+        let cart = cartProducts
+            .map {
+                POSCartItem(itemID: nil,
+                            product: $0.item,
+                            quantity: Decimal($0.quantity))
+            }
+        defer {
+            isSyncingOrder = false
+        }
+        do {
+            isSyncingOrder = true
+            let order = try await orderService.syncOrder(cart: cart,
+                                                         order: order,
+                                                         allProducts: allItems)
+            self.order = order
+            isSyncingOrder = false
+            // TODO: this is temporary solution
```

Comment:
@jaclync Do you know if there is any P2/thread/reference to what the non-temporary solution should be? We can add a link here for context.

Reply:
Good question, this TODO comment was in trunk before this PR. I just checked the history, It looks like it's from this PR comment: https://github.com/woocommerce/woocommerce-ios/pull/13099/files#r1647370089

@bozidarsevo would you like to share a link here if there is a GitHub issue or some reference about the following plan?